### DISCUSSION TITLE
UploadFile optional with multiple parameters to API

### DISCUSSION THREAD

**Discussion by varshanasi** [fastapi/fastapi#11719] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
@app.post("/search/")
async def search_similarity(query: str, file: UploadFile | None = None):
    try:
        file_content = ""
        if file:
            # Save the file
            file_path = await save_file(file)

            # Compute file hash
            file_hash = await compute_file_hash(file)
        return file_hash
    except Exception as e:
        raise HTTPException(status_code=500, detail=f"Error: {str(e)}")
```

### Description

Open the browser and expected to see an UploadFile option:
<img width="510" alt="image" src="https://github.com/tiangolo/fastapi/assets/47718420/27744900-9231-4768-85dd-5fb6eb3806b2">
Expected to see the above with the image being optional but got the following instead:
<img width="505" alt="image" src="https://github.com/tiangolo/fastapi/assets/47718420/4e570a9e-fb8a-4f32-bc8b-0f7e7d7c65d1">
My python version was 3.0+ but also tried methods mentioned for other methods but still obtained same output with empty textbox

### Operating System

Windows

### FastAPI Version

0.111.0

### Pydantic Version

2.7.3

### Python Version

3.10.12

**Comment by JavierSanchezCastro** [comment]
Check https://github.com/tiangolo/fastapi/discussions/11356#discussioncomment-8926367 https://github.com/tiangolo/fastapi/discussions/10912#discussioncomment-8068472
